tests/suite/test_v_s_route_upstream_tls.py

```diff
@@ -0,0 +1,116 @@
+
+
+@pytest.fixture(scope="class")
+def v_s_route_secure_app_setup(request, kube_apis, v_s_route_setup) -> None:
+    """
+    Prepare a secure example app for Virtual Server Route.
+
+    1st namespace with backend1-svc and backend3-svc and deployment
+    and 2nd namespace with https backend2-svc and deployment.
+
+    :param request: internal pytest fixture
+    :param kube_apis: client apis
+    :param v_s_route_setup:
+    :return:
+    """
+    print("---------------------- Deploy a VS Route Example Application ----------------------------")
+    create_items_from_yaml(kube_apis,
+                           f"{TEST_DATA}/common/app/vsr/secure/multiple.yaml", v_s_route_setup.route_m.namespace)
+
+    create_items_from_yaml(kube_apis,
+                           f"{TEST_DATA}/common/app/vsr/secure/single.yaml", v_s_route_setup.route_s.namespace)
+
+    wait_until_all_pods_are_ready(kube_apis.v1, v_s_route_setup.route_m.namespace)
+    wait_until_all_pods_are_ready(kube_apis.v1, v_s_route_setup.route_s.namespace)
+
+    def fin():
+        print("Clean up the Application:")
+        delete_items_from_yaml(kube_apis,
+                               f"{TEST_DATA}/common/app/vsr/secure/multiple.yaml",
+                               v_s_route_setup.route_m.namespace)
+        delete_items_from_yaml(kube_apis,
+                               f"{TEST_DATA}/common/app/vsr/secure/single.yaml",
+                               v_s_route_setup.route_s.namespace)
+
+    request.addfinalizer(fin)
+
+
+@pytest.mark.parametrize('crd_ingress_controller, v_s_route_setup',
+                         [({"type": "complete", "extra_args": [f"-enable-custom-resources"]},
+                           {"example": "virtual-server-route-upstream-tls"})],
+                         indirect=True)
+class TestVSRouteUpstreamOptions:
+    def test_responses_and_config_after_setup(self, kube_apis, ingress_controller_prerequisites,
+                                              crd_ingress_controller, v_s_route_setup, v_s_route_secure_app_setup):
+        ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+        config = get_vs_nginx_template_conf(kube_apis.v1,
+                                            v_s_route_setup.namespace,
+                                            v_s_route_setup.vs_name,
+                                            ic_pod_name,
+                                            ingress_controller_prerequisites.namespace)
+        req_url = f"http://{v_s_route_setup.public_endpoint.public_ip}:{v_s_route_setup.public_endpoint.port}"
+        resp_1 = requests.get(f"{req_url}{v_s_route_setup.route_m.paths[0]}",
+                              headers={"host": v_s_route_setup.vs_host})
+        resp_2 = requests.get(f"{req_url}{v_s_route_setup.route_s.paths[0]}",
+                              headers={"host": v_s_route_setup.vs_host})
+
+        assert 'proxy_pass https://' in config
```

Comment:
does it matter how many `proxy_pass https://` are in the config?

Reply:
I'll improve assertions to `assert 'proxy_pass https://{exact_host_here}' in config`